plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,146 @@
+package nodeenv
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeEnvironment", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewPodNodeEnvironment(client, "")
+	})
+}
+
+const (
+	NamespaceNodeSelector = "kubernetes.io/node-selector"
```

Comment:
I think we need to have a priority list of annotations to pull from to support our downstream usage previously using openshift.io/node-selector.

Can the `podNodeEnvironment` get configured with the slice of annotations to look at?  That way when this is started in OpenShift, we can have a single admission controller that is configured to look at both "kubernetes.io/node-selector" and "openshift.io/node-selector"?